headers/util.hpp

```diff
@@ -244,37 +242,64 @@ QVector<QString> getShortArgs(const char* data) {
 }
 
 QVector<QString> getArgs(const char* data) {
-    Document d;
-    d.Parse(data);
-    QVector<QString> args{};
-    if (d.HasMember("args")) {
-        for (const auto& m : d["args"].GetArray()) {
-            args.push_back(m.GetString());
-        }
+  Document d;
+  d.Parse(data);
+  QVector<QString> args{};
+  if (d.HasMember("args")) {
+    for (const auto& m : d["args"].GetArray()) {
+      args.push_back(m.GetString());
     }
-    return args;
+  }
+  return args;
 }
 
+QList<QString> getValueArgs(const char* data, QString key) {
+  auto key_value = key.toUtf8();
+  Document d;
+  d.Parse(data);
+  QList<QString> args{};
+  for (const auto& m : d.GetObject()) {
+    auto name = m.name.GetString();
+    if (name == key.toUtf8()) {
+      if (m.value.IsArray()) {
+        for (const auto& a : m.value.GetArray()) {
+          args.push_back(a.GetString());
+        }
+      }
+    }
+  }
+  return args;
+}
 
 CommandMap getArgMap(const char* data) {
-    Document d;
-    d.Parse(data);
-    CommandMap cm{};
-    if (d.HasMember("args")) {
-        for (const auto& m : d["args"].GetObject()) {
-            cm.emplace(std::stoi(m.name.GetString()), m.value.GetString());
-        }
+  Document d;
+  d.Parse(data);
+  CommandMap cm{};
+  if (d.HasMember("args")) {
+    for (const auto& m : d["args"].GetObject()) {
+      cm.emplace(std::stoi(m.name.GetString()), m.value.GetString());
     }
-    return cm;
+  }
+  return cm;
 }
 
 ConfigJson getConfigObject(QString json_string) {
     Document d;
     d.Parse(json_string.toUtf8());
+    StringBuffer buffer;
+    Writer<StringBuffer> writer(buffer);
     std::map<QString, QString> config_map{};
     if (d.IsObject()) {
         for (const auto& m : d.GetObject()) {
+          auto type = m.value.GetType();
```

Comment:
This function parses a JSON string object and returns a map. This change adds processing for any object value that is an array, but it needs to be refactored to handle nesting (future issue). Or we switch to using QJson.